src/Parser/FeralDruid/Modules/ComboPoints/ComboPointTracker.js

```diff
@@ -0,0 +1,120 @@
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+//import ITEMS from 'common/ITEMS';
+
+const pointGeneratingAbilities = [
+  SPELLS.SHRED.id,
+  SPELLS.RAKE.id,
+  SPELLS.THRASH_FERAL.id,
+  SPELLS.CAT_SWIPE.id,
+  SPELLS.PRIMAL_FURY.id,
+  SPELLS.ASHAMANES_FRENZY.id,
+];
+
+const pointSpendingAbilities = [
+  SPELLS.RIP.id,
+  SPELLS.MAIM.id,
+  SPELLS.FEROCIOUS_BITE.id,
+];
+
+class ComboPointTracker extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  pointsGained = 0;
+  pointsWasted = 0;
+  pointsSpent = 0;
+  currentPoints = 0;
+  maxPoints = 5;
+
+  //stores number of points gained/spent/wasted per ability ID
+  gained = {};
+  spent = {};
+  wasted = {};
+
+  on_initialized() {
+    //initialize base abilities, the rest depends on talents and equip
+    pointGeneratingAbilities.forEach(x => {
+      this.gained[x] = { points: 0 };
+      this.wasted[x] = { points: 0 };
+    });
+    pointSpendingAbilities.forEach(x => this.spent[x] = { points: 0 });
+
+    const player = this.combatants.selected;
+    if (player.hasTalent(SPELLS.LUNAR_INSPIRATION_TALENT.id)) {
```

Comment:
Instead of making these exceptions, you could have them conditionally assigned and included in your original array of abilities that generate or spend points.

Reply:
This also is at least partially mine, as I have much more of these conditions in there. I didn't want the object to contain generators/spenders that the player doesn't have, hence this approach. Any suggestions how to do it better? My only guess would be to include all the abilities regardless of availability and filter out those unavailable/unused right before rendering the table

Reply:
In terms of semantics, I think it's better to view the array as containing "all abilities the class has that do x" as opposed to "all abilities the player has that do x".